sklearn/metrics/ranking.py

```diff
@@ -710,9 +714,19 @@ def label_ranking_average_precision_score(y_true, y_score):
         scores_i = y_score[i]
         rank = rankdata(scores_i, 'max')[relevant]
         L = rankdata(scores_i[relevant], 'max')
-        out += (L / rank).mean()
+        aux = (L / rank).mean()
+        if sample_weight is None:
+            out += aux
+        else:
+            aux = aux * sample_weight[i]
+            out += aux
```

Comment:
pull this out of the if so that you only have a single `if sample_weight is not None` case.